tests/frontend/test_engine.py

```diff
@@ -347,21 +346,18 @@ def test_engine_with_mocked_api_client_sample_job(self, monkeypatch):
 
         prog = sf.Program(4)
         with prog.context as q:
-            ops.S2gate(2) | [0, 2]
-            ops.S2gate(2) | [1, 3]
-            ops.Rgate(3) | 0
-            ops.BSgate() | [0, 1]
-            ops.Rgate(3) | 0
-            ops.Rgate(3) | 1
-            ops.Rgate(3) | 2
-            ops.BSgate() | [2, 3]
-            ops.Rgate(3) | 2
-            ops.Rgate(3) | 3
-            ops.MeasureFock() | [0]
-            ops.MeasureFock() | [1]
-            ops.MeasureFock() | [2]
-            ops.MeasureFock() | [3]
-
-        engine.run(prog, "SomeProg")
+            ops.S2gate(2) | (q[0], q[2])
+            ops.S2gate(2) | (q[1], q[3])
+            ops.Rgate(3) | q[0]
+            ops.BSgate() | (q[0], q[1])
+            ops.Rgate(3) | q[0]
+            ops.BSgate(3) | (q[0], q[1])
+            ops.Rgate(3) | q[2]
+            ops.BSgate() | (q[2], q[3])
+            ops.Rgate(3) | q[2]
+            ops.BSgate() | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        engine.run(prog)
```

Comment:
Not sure why this was passing before the UI change, but not after :thinking: 
In any case, it has been updated to have correct topology now.